osu.Game/Database/LegacyImporter.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+using osu.Game.IO;
+
+namespace osu.Game.Database
+{
+    /// <summary>
+    /// A class which handles importing legacy user data of a single type from osu-stable.
+    /// </summary>
+    public abstract class LegacyImporter<TModel>
```

Comment:
Potentially `LegacyModelImporter` for clarity? The next PR in this series will be `LegacyExporter` or `LegacyModelExporter` if that adds any weight.